### PR TITLE
Feature/interrupts

### LINKED ISSUE #5: Implement interrupts
Find out how exactly interrupts work in the 8080 and implement the required functions for the Emulator.

### PR DESCRIPTION
**Pull request by moorts**
Closes #5 .

I wasn't sure if I should move this to another component as was done for the different instructions. For now, I didn't because it is really just three lines of code. If you disagree with that decision, I could change that though.

### CODE REVIEW

**Review by nico1207 — APPROVED** [review]

File: emulator/src/core/emulator.rs [hunk 1]
```diff
@@ -394,6 +398,47 @@ impl Emulator {
     pub fn load_ram(&mut self, data: Vec<u8>, start: u16) {
         self.ram.load_vec(data, start)
     }
+
+    pub fn interrupt(&mut self, opcode: u8) -> EResult<()> {
+        self.execute_instruction(opcode)
+    }
+}
+
+mod instructions;
```
Comment (felixhir) [inline emulator/src/core/emulator.rs:407]: Is this a leftover line or am I missing something?

File: emulator/src/core/emulator.rs [hunk 2]
```diff
@@ -394,6 +398,47 @@ impl Emulator {
     pub fn load_ram(&mut self, data: Vec<u8>, start: u16) {
         self.ram.load_vec(data, start)
     }
+
+    pub fn interrupt(&mut self, opcode: u8) -> EResult<()> {
```
Comment (felixhir) [inline emulator/src/core/emulator.rs:402]: Maybe I am not aware enough of how interrupts work, but isn't this change just encapsulating the original `execute_next()` method? What's the reason behind this?

Comment (nico1207) [reply emulator/src/core/emulator.rs:402]: An external device calls an interrupt by sending one opcode to the CPU which is executed (typically a RST instruction).

Comment (nico1207) [inline emulator/src/core/emulator.rs:402]: Shouldn't this method also check whether interrupts are currently enabled? (EI and DI instructions)

**Review by nico1207 — CHANGES_REQUESTED** [review]

File: emulator/src/core/emulator.rs (hunk 2 above)
Comment (nico1207) [reply emulator/src/core/emulator.rs:402]: and also reset the "INTE" bit after executing the instruction

Comment (felixhir) [reply emulator/src/core/emulator.rs:402]: But doesn't this method call do the same as the method we used earlier? I'm only wondering if this is basically a refactor to better match the naming.

Comment (nico1207) [reply emulator/src/core/emulator.rs:402]: "Whenever INTE is equal to 0, the entire interrupt handling system is disabled, and no interrupts will be accepted."

"When the CPU recognizes an interrupt request from an
external device, the following actions occur:
(1) The instruction currently being executed is completed.
(2) The interrupt enable bit, INTE, is reset = O.
(3) The interrupting device supplies, via hardware, one instruction which the CPU executes."

Comment (nico1207) [reply emulator/src/core/emulator.rs:402]: The method will also need to do some other stuff (see my change request)

**Review by nico1207 — APPROVED** [review]

**Pull request merged**